src/compiler/checker.ts

```diff
@@ -13255,23 +13255,68 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
         return result;
     }
 
+    function getFallbackExpandedLabels(restType: TupleTypeReference, restDeclaration: Declaration | undefined) {
+        if (!restDeclaration) {
+            return;
+        }
+        Debug.assert(isParameter(restDeclaration));
+        if (restDeclaration.name.kind !== SyntaxKind.ArrayBindingPattern) {
+            return;
+        }
+        const nameElements = restDeclaration.name.elements;
+        const elementFlags = restType.target.elementFlags;
+        const length = Math.min(nameElements.length, elementFlags.length);
+        const result: (__String | undefined)[] = [];
+        for (let i = 0; i < length; i++) {

```

Comment:
I can't really put my finger on why this feels like a lot of code, but it feels like this would be simpler if this just preallocated an array of undefineds and then wrote into it by index.